code/modules/reagents/Chemistry-Reagents/Toxin-Reagents.dm

```diff
@@ -120,6 +121,527 @@ datum/reagent/toxin/slimejelly/on_mob_life(var/mob/living/M as mob)
+			A.xo = 0
+	A.fire()
+	holder.remove_reagent(src.id,10,safety = 1)
+datum/reagent/toxin/emit_on/on_mob_life(var/mob/living/M as mob)
+	M.adjustFireLoss(10)
+	var/emit_dir = pick(1,2,3,4)
+	var/obj/item/projectile/beam/emitter/A = PoolOrNew(/obj/item/projectile/beam/emitter,get_turf(holder.my_atom))
+	switch(emit_dir)
+		if(1)
+			A.yo = 20
+			A.xo = 0
+		if(2)
+			A.yo = 0
+			A.xo = 20
+		if(3)
+			A.yo = 0
+			A.xo = -20
+		else
+			A.yo = -20
+			A.xo = 0
+	A.fire()
+	holder.remove_reagent(src.id,10,safety = 1)
+datum/reagent/toxin/emit_on/on_fart(var/mob/living/M as mob)
+	M.adjustFireLoss(50)
+	var/obj/item/projectile/beam/emitter/A = PoolOrNew(/obj/item/projectile/beam/emitter,get_turf(holder.my_atom))
+	var/obj/item/projectile/beam/emitter/B = PoolOrNew(/obj/item/projectile/beam/emitter,get_turf(holder.my_atom))
+	var/obj/item/projectile/beam/emitter/C = PoolOrNew(/obj/item/projectile/beam/emitter,get_turf(holder.my_atom))
+	var/obj/item/projectile/beam/emitter/D = PoolOrNew(/obj/item/projectile/beam/emitter,get_turf(holder.my_atom))
+	A.yo = 20
+	A.xo = 0
+	B.yo = 0
+	B.xo = 20
+	C.yo = 0
+	C.xo = -20
+	D.yo = -20
+	D.xo = 0
+	A.fire()
+	B.fire()
+	C.fire()
+	D.fire()
+	holder.remove_reagent(src.id,100,safety = 1)
+datum/reagent/toxin/cryonium
+	name = "Cryonium"
+	id = "cryo"
+	description = "Chilly."
+	color = "#75AC53"
+datum/reagent/toxin/cryonium/on_mob_life(var/mob/living/M as mob)
+	M.bodytemperature -= 3//this is how you avoid super boom
+datum/reagent/toxin/cryonium/on_update()
+	if(!holder)
+		return
+	if(holder.present_machines[1]-5 >= 0)
+		holder.present_machines[1] -= 5
+	else
+		holder.present_machines[1] -= (5- holder.present_machines[1])
+	holder.remove_reagent(src.id,5)
+datum/reagent/toxin/pyrosium
+	name = "Pyrosium"
+	id = "pyro"
+	description = "Warm."
+	color = "#75AC53"
+datum/reagent/toxin/pyrosium/on_mob_life(var/mob/living/M as mob)
+	M.bodytemperature += 3
+	holder.remove_reagent(src.id,5,safety = 1)
+datum/reagent/toxin/pyrosium/on_update()
+	if(!holder)
+		return
+	if(holder.present_machines[1]+5 <= 500)
+		holder.present_machines[1] += 5
+	else
+		holder.present_machines[1] += (500- holder.present_machines[1])
+	holder.remove_reagent(src.id,5)
+datum/reagent/toxin/emote
+	name = "Pure Emotium"
+	id = "emote"
+	description = "This shouldn't be difficult to figure out."
+	color = "#75AC53"
+datum/reagent/toxin/emote/on_mob_life(var/mob/living/M as mob)
+	if(prob(20))
+		M.emote(pick("fart","flap","aflap","airguitar","blink","shrug","cough","sneeze","shake","twitch"))
+		M.hallucination += 1
+datum/reagent/toxin/smokep
+	name = "Smoke Powder"
+	id = "smokep"
+	description = "Try farting I dare you!"
+	color = "#75AC53"
+datum/reagent/toxin/smokep/on_mob_life(var/mob/living/M as mob)
+	if(prob(5))
+		M.emote(pick("cough","sneeze"))
+datum/reagent/toxin/smokep/on_fart(var/mob/living/M as mob)
+	M.visible_message("<span class='danger'>[M] farts out a noxious cloud!</span>")
+	create_smoke(src, 10)
```

Comment:
this isn't a reagents holder you're feeding in, you're feeding in /datum/reagent

this will runtime badly